server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlDialect.java

```diff
@@ -89,7 +92,7 @@ public ResultWithErrors<DialectOutcome> processText(DialectProcessingContext con
     nodes.addAll(context.getDialectNodes());
 
     // Add error encountered while visiting the parser. To be reported to COBOL LS engine.
-    parseError.addAll(db2SqlVisitor.getErrors());
+    if (isSqlProcessingEnabled) parseError.addAll(db2SqlVisitor.getErrors());
```

Comment:
Don't produce the errors in the first place.
```suggestion
    parseError.addAll(db2SqlVisitor.getErrors());
```